meson.build

```diff
@@ -0,0 +1,624 @@
+	if have == true
+		conf.set10('HAVE_' + ident[0].to_upper(), have)
+	endif
+endforeach
+
+threads = dependency('threads')
+libseccomp = cc.find_library('seccomp', required : false)
+if libseccomp.found()
+	conf.set10('HAVE_SECCOMP', libseccomp.found())
+
+	seccomp_headers = '''
+	#include <seccomp.h>
+	'''
+
+	if cc.has_function('seccomp_notify_fd', prefix : seccomp_headers, args : '-D_GNU_SOURCE', dependencies: libseccomp)
+	        conf.set10('HAVE_DECL_SECCOMP_NOTIFY_FD', true)
+	endif
+
+	if cc.has_function('seccomp_syscall_resolve_name_arch', prefix : seccomp_headers, args : '-D_GNU_SOURCE', dependencies: libseccomp)
+	        conf.set10('HAVE_DECL_SECCOMP_SYSCALL_RESOLVE_NAME_ARCH', true)
+	endif
+
+	foreach decl : [
+	                'scmp_filter_ctx',
+	                'struct seccomp_notif_sizes',
+	                'struct clone_args',
+	               ]
+
+	        # We get -1 if the size cannot be determined
+	        have = cc.sizeof(decl, prefix : seccomp_headers, args : '-D_GNU_SOURCE') > 0
+
+		if have == true
+			conf.set10('HAVE_' + decl.underscorify().to_upper(), have)
+		endif
+	endforeach
+endif
+
+libselinux = cc.find_library('selinux', required : false)
+if libselinux.found()
+	conf.set10('HAVE_SELINUX', libselinux.found())
+endif
+
+libapparmor = cc.find_library('apparmor', required : false)
+if libapparmor.found()
+	conf.set10('HAVE_APPARMOR', libapparmor.found())
+endif
+
+libopenssl = dependency('openssl', required : false)
+if libopenssl.found()
+	conf.set10('HAVE_OPENSSL', libopenssl.found())
+endif
+
+libcap = dependency('libcap', required : false)
+if not libcap.found()
+        # Compat with Ubuntu 14.04 which ships libcap w/o .pc file
+        libcap = cc.find_library('cap')
+endif
+
+if libcap.found()
+	conf.set10('HAVE_LIBCAP', libcap.found())
+endif
+
+basic_includes = include_directories(
+	'src',
+	'src/include',
+	'.')
+
+liblxc_includes = [basic_includes, include_directories(
+        'src/lxc/cgroups',
+        'src/lxc/lsm',
+        'src/lxc/storage')]
+
+add_project_arguments('-include', 'config.h', language : 'c')
+
+subdir('src/include')
+subdir('src/lxc')
+subdir('src/lxc/tools')
+
+config_h = configure_file(
+        output : 'config.h',
+        configuration : conf)
+
+liblxc = shared_library(
+        'lxc',
+        version : liblxc_version,
+        include_directories : tools_liblxc_includes,
+        link_args : ['-shared', '-fPIC', '-DPIC'],
+        link_whole : [liblxc_static],
+        dependencies : [threads,
+			libseccomp,
+			libcap,
+			libopenssl,
+			libselinux,
+			libapparmor],
+        install : true,
+        install_dir : libdir)
+
+public_programs = []
+
+public_programs += executable(
+                'lxc-autostart',
+		tools_lxc_autostart_sources,
+                include_directories : tools_liblxc_includes,
+                link_with : [liblxc],
+		dependencies : [threads,
+				libseccomp,
+				libcap,
+				libopenssl,
+				libselinux,
+				libapparmor],
```

Comment:
Since you link with liblxc for many executables, and also a list of the same dependencies each time, you could consider using

```
liblxc_dep = declare_dependency(
    link_with: liblxc,
    dependencies: [threads,
                  libseccomp,
                  libcap,
                  libopenssl,
                  libselinux,
                  libapparmor])
```

Then each executable can `dependencies: liblxc_dep` and it expands to all these repetitive arguments.

Reply:
done